lib/fluent/plugin/filter_parser.rb

```diff
@@ -0,0 +1,143 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'fluent/time'
+require 'fluent/config/error'
+require 'fluent/plugin/filter'
+require 'fluent/plugin_helper/parser'
+require 'fluent/plugin_helper/compat_parameters'
+
+module Fluent::Plugin
+  class ParserFilter < Filter
+    Fluent::Plugin.register_filter('parser', self)
+
+    helpers :parser, :compat_parameters
+
+    config_param :key_name, :string
+    config_param :reserve_data, :bool, default: false
+    config_param :inject_key_prefix, :string, default: nil
+    config_param :replace_invalid_sequence, :bool, default: false
+    config_param :hash_value_field, :string, default: nil
+    config_param :suppress_parse_error_log, :bool, default: false
+    config_param :time_parse, :bool, default: true
+    config_param :ignore_key_not_exist, :bool, default: false
+
+    attr_reader :parser
+
+    def configure(conf)
+      compat_parameters_convert(conf, :parser)
+      parser_config = conf.elements('parse').first
+      unless parser_config
+        raise Fluent::ConfigError, "<parse> section is required"
+      end
+      unless parser_config["@type"]
+        raise Fluent::ConfigError, "parse/@type is required"
+      end
+
+      super
+
+      @parser = parser_create(usage: 'filter_parser', conf: parser_config)
```

Comment:
But now, I'm thinking it's better NOT to set `@parser.time_key` anytime, and:

* rename `time_parse` configuration option to `reserve_time` (consistent with `reserve_data`)
* ignore parsed time (from `@parser.parse`) when `@reserve_time` is true

It makes the plugin code highly cleaner and simpler.
Parser plugins have these own default values for `time_key`, and I think it's better not to touch them from parser plugin.